documentation/demo.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * To run this example, you must
+ * 1. git checkout https://github.com/daostack/client
+ * 2. npmm install
+ * 3. npm run build // build th epackage
+ * 3. docker-compose up -d // this will start docker containers with the test services
+ * 4. node documentation/demo.js // run this file
+ * 5. docker-compose down // stop the docker containers when you are done
+ */
+
+async function main() {
+
+  console.log('hello!')
+
+  // mport from the local build of the library (that was created with npm run build)
+  // typically, one would use require('@daostack/client')
+  const { Arc } = require('../dist/lib/index.js')
+
+  // "Arc" is the main class that handles configuration and connections to various services
+  // create an Arc instanc with settings to connect to the local docker images
+  const arc = new Arc({
+    graphqlHttpProvider: 'http://127.0.0.1:8000/subgraphs/name/daostack',
+    graphqlHttpMetaProvider: 'http://127.0.0.1:8000/subgraphs',
+    graphqlWsProvider: 'http://127.0.0.1:8001/subgraphs/name/daostack',
+    web3Provider: 'ws://127.0.0.1:8545',
+    ipfsProvider: '/ip4/127.0.0.1/tcp/5001',
+  })
+  // we must provice Arc with some contract information.
+  // We could have hardcoded it, but get this information from the subgraph
+  const contractInfos = await arc.fetchContractInfos()
+  arc.setContractInfos(contractInfos)
```

Comment:
Is the extra call to `setContractInfos` needed, or should it just set it automatically within `fetchContractInfos`?

Reply:
I am not sure, perhaps yes.

Reply:
ok yes :-)